components/mailchimp/sources/new-or-updated-subscriber/new-or-updated-subscriber.js

```diff
@@ -0,0 +1,42 @@
+const common = require("../common/http-based");
+
+module.exports = {
+  ...common,
+  key: "new-or-updated-subscriber",
+  name: "New Or Updated Subscriber",
+  description:
+    "Emit new event when a subscriber is added or updated (on profile, or email address change) in an audience list.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+  },
+  methods: {
+    ...common.methods,
+    getEventTypes() {
+      return [
+        "subscribe",
+        "profile",
+        "upemail",
+        "unsubscribe",
+        "cleaned",
+      ];
+    },
+    generateMeta(eventPayload) {
+      const ts = Date.parse(eventPayload.fired_at);
+      let summary;
+      if (this.getEventTypes().includes(eventPayload.type)) {
+        summary =`New event "${eventPayload.type}" occurred`;
+      } else {
+        console.log("Unexpected trigger found, skipping emit");
+        return;
+      }
```

Comment:
Could we write it this way?
```suggestion
      if (!this.getEventTypes().includes(eventPayload.type)) {
        console.log("Unexpected trigger found, skipping emit");
        return;
      }
      const summary =`New event "${eventPayload.type}" occurred`;
      return {
        id: eventPayload["data[id]"],
        summary,
        ts,
      };
```